plugins/modules/packaging/os/portage.py

```diff
@@ -262,7 +266,12 @@ def query_set(module, set, action):
             module.fail_json(msg='set %s cannot be removed' % set)
         return False
 
-    world_sets_path = '/var/lib/portage/world_sets'
+    try:
+        # When providing ROOT via ansible enviroment, portage.root may return a
+        # portage._LegacyGlobalProxy instance. Force it to be a string.
+        world_sets_path = os.path.join(str(portage.root), portage.const.WORLD_SETS_FILE)
+    except NameError:
+        world_sets_path = "/var/lib/portage/world_sets"
```

Comment:
Nit: changing the quotation style unnecessarily makes the diff harder to read, try to avoid such unrelated changes to make it easier for the reviewers to catch up with what's going on in the PR.